pvlib/iotools/ecmwf_macc.py

```diff
@@ -25,7 +25,10 @@ def ECMWFDataServer(*a, **kw):
         )
     SERVER = None
 else:
-    SERVER = ECMWFDataServer()
+    try:
+        SERVER = ECMWFDataServer()
+    except KeyError:
```

Comment:
Is there an informative message we could raise?  Just a thought.

Reply:
As a warning? okay, sounds good to me. Any objections?

Reply:
I like the idea of printing a warning here (`ecmwfapi` installed, but not properly configured).

Reply:
Is it really necessary to run this function during the import?  Or can it be made unnecessary?

Reply:
No, it's not really necessary to run this function during import.

If `SERVER` is set to `None`, then it will be called later in [`get_ecmwf_macc()`](https://github.com/pvlib/pvlib-python/blob/master/pvlib/iotools/ecmwf_macc.py#L169). I am open to removing it completely from the top level of the module, and letting `get_ecmwf_macc()` create a new server for every call. It's not an expensive call, and I doubt users are making many calls to get ECMWF data. Since `get_ecmwf_macc()` takes `server` as an argument, if a user plans to make a lot of calls, then they can call `ecmwfapi.ECMWFDataServer()` separately and pass it as an argument.

Is this your preference?